src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -115,6 +126,12 @@ def _extract_messages(self) -> None:
             assert isinstance(msg.value, BrokerValue)
             partition_offset = PartitionIdxOffset(msg.value.partition.index, msg.value.offset)
 
+            if namespace := self._extract_namespace(msg.payload.headers) in options.get(
```

Comment:
We should abstract this out so it only applies to the generic metrics pipeline. Since this code isn't applicable for release health pipeline, no need for that pipeline to go through this

Reply:
I am more inclined to keep it regardless of the pipeline if it does not cause any performance regression. Because we want the kill switch to work on all use case IDs, it would be jarring if someone set the `disabled-namespaces` to `['spans', 'sessions']` only for spans to be intercepted but sessions continues to work normally.

Reply:
Is use case awareness something we want to expose as a feature on the release health pipeline? I would imagine not. There are these subtle differences between the the pipelines and it would be nice if those differences were captured in a more concrete manner. Having same same level of flexibility across pipelines though nice to have can also make things hard to refactor.

My attempt at handling some of these differences are captured in this [PR](https://github.com/getsentry/sentry/pull/53493). Happy to discuss more if you want to.